torch/distributed/tensor/_ops/_math_ops.py

```diff
@@ -1144,6 +1144,35 @@ def topk_strategy(op_schema: OpSchema) -> OpStrategy:
     )
 
 
+@register_op_strategy(
+    [aten.sort.stable, aten.sort.default],
+    schema_info=RuntimeSchemaInfo(
+        1,
+        static_kwargkey=["dim", "descending", "stable"],
+    ),
+)
+def sort_strategy(op_schema: OpSchema) -> OpStrategy:
+    # mostly copy past from topk_strategy
+    input_strategy = op_schema.args_schema[0]
+    assert isinstance(input_strategy, OpStrategy)
+    dim = -1
+    if "dim" in op_schema.kwargs_schema:
+        dim = cast(int, op_schema.kwargs_schema["dim"])
+    elif len(op_schema.args_schema) > 1:
+        dim = cast(int, op_schema.args_schema[1])
```

Comment:
you introduced this because these 2 ops have different op signature. I suggest we separate. The common logic can be extracted and reused, as a helper function if not an all-replicate util